src/mrinufft/operators/interfaces/cufinufft.py

```diff
@@ -87,13 +85,15 @@ def _make_plan(self, typ, **kwargs):
             **kwargs,
         )
 
-    def _set_pts(self, typ, samples):
+    def _set_kxyz(self, samples):
+        self._kx.set(samples[:, 0])
+        self._ky.set(samples[:, 1])
+        if self.ndim == 3:
+            self._kz.set(samples[:, 2])
+
+    def _set_pts(self, typ):
         plan = self.grad_plan if typ == "grad" else self.plans[typ]
-        plan.setpts(
-            cp.array(samples[:, 0], copy=False),
-            cp.array(samples[:, 1], copy=False),
-            cp.array(samples[:, 2], copy=False) if self.ndim == 3 else None,
-        )
+        plan.setpts(self._kx, self._ky, self._kz)
```

Comment:
Seems redundant?

Reply:
Recommendation to have it better?

Reply:
well I don't understand why you need `_set_pts` and `_set_kxyz` for me `_set_pts(self, typ, samples)` should be enough. keeping an attribute `self._samples` would also be simpler (IMO) than `_kx,_ky,_kz`.